src/hammer-vlsi/synthesis/vivado/file_templates/vivado_setup.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# sample vivado setup script.
+
+# setup license server
+source /tools/flexlm/flexlm.sh
+# setup vivado environment variables
+source /tools/xilinx/Vivado/2018.1/settings64.sh
```

Comment:
The setup script seems obsolete since we have synthesis.vivado.binary. We should also specify as much as possible the license variables in the defaults.yml to avoid over reliance on CAD shell scripts

Reply:
Do you mean you want to parse through those scripts (which ultimately just sets various environment variables), and list all those variables in defaults.yml instead?  Or something else?  I don't think having synthesis.vivado.binary is sufficient, as Vivado may need some environment variables to be set by the shell script to function.

In general, I find the approach of asking user to provide a "master shell script" which sets up everything needed by the tool (including licenses) to be most portable across different setups.  As a very contrived example, if someone in Xilinx is trying to use Hammer with a developer version of Vivado, he may not need to set up licenses at all.

What do you think?

Reply:
Long-form thoughts below